molgenis-data/src/main/java/org/molgenis/data/support/DefaultAttributeMetaData.java

```diff
@@ -23,8 +47,10 @@
  */
 public class DefaultAttributeMetaData implements AttributeMetaData
 {
-	private String name;
-	private FieldType fieldType = MolgenisFieldTypes.STRING;
+	private Map<String, AttributeChangeListener> changeListeners;
+
+	private final String name;
+	private FieldType fieldType;
 	private String description;
 	private boolean nillable = true;
 	private boolean readOnly = false;
```

Comment:
Can you check the 
private boolean labelAttribute = false; // remove?
private boolean lookupAttribute = false; // remove?

If we can remove those params than, you can remove some of your code.